raiden/tests/scenarios/long-running.yaml

```diff
@@ -0,0 +1,319 @@
+            - assert: {from: 3, to: 2, total_deposit: 101, balance: 101, state: "opened"}
+      - serial:
+          name: "C tries different cases"
+          tasks:
+            - open_channel: {from: 2, to: 4, total_deposit: 0, expected_http_status: 201}
+            - wait: 10
+            - deposit: {from: 2, to: 4, total_deposit: 100}
+            - wait: 10
+            # Try to make a deposit smaller the the amount that was deposited
+            - deposit: {from: 2, to: 4, total_deposit: 30, expected_http_status: 409}
+            # Make a deposit bigger then the deposited amount
+            - deposit: {from: 2, to: 4, total_deposit: 150}
+            - wait: 10
+      - parallel:
+          name: "Verify the channels"
+          tasks:
+            - assert: {from: 0, to: 1, total_deposit: 100, balance: 100, state: "opened"}
+            - assert: {from: 1, to: 0, total_deposit: 100, balance: 100, state: "opened"}
+            - assert: {from: 1, to: 2, total_deposit: 100, balance: 100, state: "opened"}
+            - assert: {from: 2, to: 1, total_deposit: 100, balance: 100, state: "opened"}
+            - assert: {from: 2, to: 3, total_deposit: 101, balance: 101, state: "opened"}
+            - assert: {from: 3, to: 2, total_deposit: 101, balance: 101, state: "opened"}
+            # New channel
+            - assert: {from: 2, to: 4, total_deposit: 150, balance: 150, state: "opened"}
+            - assert: {from: 4, to: 2, total_deposit: 0, balance: 0, state: "opened"}
+      - serial:
+          name: "Continue building the network"
+          tasks:
+            # A opens a channel with E
+            - open_channel: {from: 0, to: 4, total_deposit: 10, expected_http_status: 201}
+            - assert: {from: 0, to: 4, total_deposit: 10, balance: 10, state: "opened"}
+            # E deposits 25 tokens onto the A - E channel
+            - deposit: {from: 4, to: 0, total_deposit: 25}
+            - wait: 20
+            # A performs a payment to E of 50 tokens (path A<->B<->C<->E)
+            - transfer: {from: 0, to: 4, amount: 50, expected_http_status: 200}
+            - wait: 5
+            - assert: {from: 0, to: 1, total_deposit: 100, balance: 50, state: "opened"}
+            - assert: {from: 1, to: 0, total_deposit: 100, balance: 150, state: "opened"}
+            - assert: {from: 1, to: 2, total_deposit: 100, balance: 50, state: "opened"}
+            - assert: {from: 2, to: 1, total_deposit: 100, balance: 150, state: "opened"}
+            - assert: {from: 2, to: 4, total_deposit: 150, balance: 100, state: "opened"}
+            - assert: {from: 4, to: 2, total_deposit: 0, balance: 50, state: "opened"}
+            # C sends all of its tokens to B (one transfer)
+            - transfer: {from: 2, to: 1, amount: 150, expected_http_status: 200}
+            - wait: 5
+            - assert: {from: 2, to: 1, total_deposit: 100, balance: 0, state: "opened"}
```

Comment:
As explained in the call we had, this means that:
1. node (2) has a direct channel to node (1)
2. node (2) has 0 balance left to transfer

Reply:
Before line 117 the network looks like this:

```
Node(0) <------------------------> Node(1)
  ^                                  ^                                    
  |                                  |   
  |                                  |   
  |                                  |   
  |                                  |   
  |                                  |   
 \/                                  \/   
Node(4) <------------------------> Node(2)
                                     ^
                                     |
Node(3) <-----------------------------
```

With the following channel capacities:


|   Node  | Partner | Capacity |
|:-------:|---------|----------|
| Node(0) | Node(1) |    100   |
| Node(0) | Node(4) |    10    |
| Node(1) | Node(0) |    100   |
| Node(1) | Node(2) |    100   |
| Node(2) | Node(1) |    100   |
| Node(2) | Node(4) |    150   |
| Node(2) | Node(3) |    101   |
| Node(3) | Node(2) |    101   |
| Node(4) | Node(0) |    25    |
| Node(4) | Node(2) |    0     |


I belive your argument is wrong and let's take line 117<Insert line here> as an example. We have the
channel ``Node(0)`` <-> ``Node(4)``, but the channel does not have the required capacity for the 
payment to be performed so it choses the other route which is through 
``Node(0)`` <-> ``Node(1)`` <-> ``Node(2)`` <-> ``Node(4)`` (this was also mentioned down).



The capacity of each channel on the network will look like this afterwords:

|   Node  | Partner | Capacity |
|:-------:|---------|----------|
| Node(0) | Node(1) |    50    |
| Node(0) | Node(4) |    10    |
| Node(1) | Node(0) |    150   |
| Node(1) | Node(2) |    50    |
| Node(2) | Node(1) |    150   |
| Node(2) | Node(4) |    100   |
| Node(2) | Node(3) |    101   |
| Node(3) | Node(2) |    101   |
| Node(4) | Node(0) |    25    |
| Node(4) | Node(2) |    50    |


As you mentioned then the payment 

``- transfer: {from: 2, to: 1, amount: 150, expected_http_status: 200}`` is performed. And the 
channel capacity will look like this.


|   Node  | Partner | Capacity |
|:-------:|---------|----------|
| Node(0) | Node(1) |    50    |
| Node(0) | Node(4) |    10    |
| Node(1) | Node(0) |    150   |
| Node(1) | Node(2) |    200   |
| Node(2) | Node(1) |    0     |
| Node(2) | Node(4) |    100   |
| Node(2) | Node(3) |    101   |
| Node(3) | Node(2) |    101   |
| Node(4) | Node(0) |    25    |
| Node(4) | Node(2) |    50    |


Which leaves the channel ``Node(2)`` <-> ``Node(1)`` with 0 capacity, as you mentioned. 

Next is the ``- transfer: {from: 2, to: 1, amount: 150, expected_http_status: 409}``
which works as expected because there is no capacity neither in the ``Node(2) <-> ``Node(1)`` channel
nor in the ``Node(2) <-> Node(4) <-> Node(0) <-> Node(1)``. 

But the transfer ``- transfer: {from: 2, to: 1, amount: 1}`` should work, beacuse event though 
``Node(2) <-> Node(1)`` route doesn't have the required capacity, the other route availabel has the required capacity
``Node(2) <-> Node(4) <-> Node(0) <-> Node(1)``.


This is the same as the transfer ``from 0 to 4`` where there was no capacity in the 
``Node(0)`` <-> ``Node(4)`` but enough on the other route: 
``Node(0) <-> Node(1) <-> Node(2) <-> Node(4)`` because it has the capacity. 







I've also build a scenario to prove that this was the expected behavior: 


This is how the network will look like after line 55 where we try to perform payments from 0 to 1
using the ``0 -> 3 -> 2 -> 1`` route.

```
Line 56:
      - serial:
          name: "0 performs payments to 1 using the route 0->3->2->1"
          repeat: 20
          tasks:
            - transfer: {from: 0, to: 1, amount: 1}
            - wait: 5
```


```
Node(0) <-> Node(1) <-> Node(2) <-> Node(3)
  ^				         ^	
  |				         |
   ------------------------------------
```

And the capacity of each channels:

|   Node  | Partner | Capacity |
|:-------:|---------|----------|
| Node(0) | Node(1) |     0    |
| Node(1) | Node(0) |     0    |
| Node(1) | Node(2) |    100   |
| Node(2) | Node(1) |    100   |
| Node(2) | Node(3) |    100   |
| Node(3) | Node(2) |    100   |
| Node(0) | Node(3) |    100   |
| Node(3) | Node(0) |    100   |



```
version: 2

settings:
  gas_price: "fast"
  chain: kovan

token:
  decimals: 3

nodes:
  mode: managed
  count: 5
  raiden_version: /home/cosmin/Documents/utility/mainnet_raiden/1002rc3/raiden-v0.100.2rc3.dev2+g04356e2c-linux

  default_options:
    gas-price: fast
    environment-type: development


scenario:
  serial:
    name: "Routing scenario"
    tasks:
      - parallel:
          name: "Create channels"
          tasks:
            - open_channel: {from: 0, to: 1, total_deposit: 0, expected_http_status: 201}
            - open_channel: {from: 1, to: 2, total_deposit: 100, expected_http_status: 201}
            - open_channel: {from: 2, to: 3, total_deposit: 100, expected_http_status: 201}
            - open_channel: {from: 0, to: 3, total_deposit: 100, expected_http_status: 201}
      - serial:
          name: "Allow time for nodes to learn about their channels"
          tasks:
            - wait: 30
      - parallel:
          name: "Founding the channles"
          tasks:
            - deposit: {from: 2, to: 1, total_deposit: 100}
            - deposit: {from: 3, to: 2, total_deposit: 100}
            - deposit: {from: 3, to: 0, total_deposit: 100}
      - serial:
          name: "Allow time for all nodes to learn about the deposit"
          tasks:
            - wait: 15
      - parallel:
          name: "Checking if the deposits were successful"
          tasks:
            - assert: {from: 0, to: 1, total_deposit: 0, balance: 0, state: "opened"}
            - assert: {from: 1, to: 0, total_deposit: 0, balance: 0, state: "opened"}
            - assert: {from: 1, to: 2, total_deposit: 100, balance: 100, state: "opened"}
            - assert: {from: 2, to: 1, total_deposit: 100, balance: 100, state: "opened"}
            - assert: {from: 2, to: 3, total_deposit: 100, balance: 100, state: "opened"}
            - assert: {from: 3, to: 2, total_deposit: 100, balance: 100, state: "opened"}
            - assert: {from: 0, to: 3, total_deposit: 100, balance: 100, state: "opened"}
            - assert: {from: 3, to: 0, total_deposit: 100, balance: 100, state: "opened"}
      - serial:
          name: "0 performs payments to 1 using the route 0->3->2->1"
          repeat: 20
          tasks:
            - transfer: {from: 0, to: 1, amount: 1}
            - wait: 5
      - parallel:
          name: "Checking if the transfers were executed corectly"
          tasks:
            - assert: {from: 0, to: 1, total_deposit: 0, balance: 0, state: "opened"}
            - assert: {from: 1, to: 0, total_deposit: 0, balance: 0, state: "opened"}
            - assert: {from: 1, to: 2, total_deposit: 100, balance: 120, state: "opened"}
            - assert: {from: 2, to: 1, total_deposit: 100, balance: 80, state: "opened"}
            - assert: {from: 2, to: 3, total_deposit: 100, balance: 120, state: "opened"}
            - assert: {from: 3, to: 2, total_deposit: 100, balance: 80, state: "opened"}
            - assert: {from: 0, to: 3, total_deposit: 100, balance: 80, state: "opened"}
            - assert: {from: 3, to: 0, total_deposit: 100, balance: 120, state: "opened"}
      - parallel:
          name: "Founding the channles"
          tasks:
            - deposit: {from: 0, to: 1, total_deposit: 100}
            - deposit: {from: 1, to: 0, total_deposit: 100}
      - serial:
          name: "Allow time for nodes 0 and 1 to learn about the deposit"
          tasks:
            - wait: 15
      - parallel:
          name: "Checking if the transfers were executed corectly"
          tasks:
            - assert: {from: 0, to: 1, total_deposit: 100, balance: 100, state: "opened"}
            - assert: {from: 1, to: 0, total_deposit: 100, balance: 100, state: "opened"}
            - assert: {from: 1, to: 2, total_deposit: 100, balance: 120, state: "opened"}
            - assert: {from: 2, to: 1, total_deposit: 100, balance: 80, state: "opened"}
            - assert: {from: 2, to: 3, total_deposit: 100, balance: 120, state: "opened"}
            - assert: {from: 3, to: 2, total_deposit: 100, balance: 80, state: "opened"}
            - assert: {from: 0, to: 3, total_deposit: 100, balance: 80, state: "opened"}
            - assert: {from: 3, to: 0, total_deposit: 100, balance: 120, state: "opened"}
            - assert: {from: 0, to: 1, total_deposit: 100, balance: 100, state: "opened"}
            - assert: {from: 1, to: 0, total_deposit: 100, balance: 100, state: "opened"}
      - serial:
          name: "0 performs payments to 1 using their channel"
          repeat: 20
          tasks:
            - transfer: {from: 0, to: 1, amount: 1}
            - wait: 5
      - parallel:
          name: "Checking if the transfers were executed corectly"
          tasks:
            - assert: {from: 0, to: 1, total_deposit: 100, balance: 80, state: "opened"}
            - assert: {from: 1, to: 0, total_deposit: 100, balance: 120, state: "opened"}
            - assert: {from: 1, to: 2, total_deposit: 100, balance: 120, state: "opened"}
            - assert: {from: 2, to: 1, total_deposit: 100, balance: 80, state: "opened"}
            - assert: {from: 2, to: 3, total_deposit: 100, balance: 120, state: "opened"}
            - assert: {from: 3, to: 2, total_deposit: 100, balance: 80, state: "opened"}
            - assert: {from: 0, to: 3, total_deposit: 100, balance: 80, state: "opened"}
            - assert: {from: 3, to: 0, total_deposit: 100, balance: 120, state: "opened"}
      - serial:
          name: "Closing the 0-1 channel"
          tasks:
            - close_channel: {from: 0, to: 1}
            - wait: 20
      - serial:
          name: "0 performs payments to 1 using the route 0->3->2->1"
          repeat: 20
          tasks:
            - transfer: {from: 0, to: 1, amount: 1}
            - wait: 5
      - parallel:
          name: "Checking if the transfers were executed corectly"
          tasks:
            - assert: {from: 0, to: 1, total_deposit: 100, balance: 80, state: "closed"}
            - assert: {from: 1, to: 0, total_deposit: 100, balance: 120, state: "closed"}
            - assert: {from: 1, to: 2, total_deposit: 100, balance: 140, state: "opened"}
            - assert: {from: 2, to: 1, total_deposit: 100, balance: 60, state: "opened"}
            - assert: {from: 2, to: 3, total_deposit: 100, balance: 140, state: "opened"}
            - assert: {from: 3, to: 2, total_deposit: 100, balance: 60, state: "opened"}
            - assert: {from: 0, to: 3, total_deposit: 100, balance: 60, state: "opened"}
            - assert: {from: 3, to: 0, total_deposit: 100, balance: 140, state: "opened"}
```